services/priceService.ts

```diff
@@ -0,0 +1,159 @@
+  const isoString = (new Date(timestamp * 1000)).toISOString()
+  return isoString.split('T')[0].replace(/-/g, '') // YYYYMMDD
+}
+
+export async function syncPricesFromTransactionList (transactions: TransactionWithAddressAndPrices[]): Promise<void> {
+  for (const transaction of transactions) {
+    void await syncTransactionPrices({
+      networkId: transaction.address.networkId,
+      timestamp: transaction.timestamp,
+      transactionId: transaction.id
+    })
+  }
+}
+export interface CreateAllPricesFromTransactionInput {
+  timestamp: number
+  networkId: number
+  transactionId: number
+  values: {
+    'usd': Prisma.Decimal
+    'cad': Prisma.Decimal
+  }
+}
+
+export interface SyncTransactionPricesInput {
+  networkId: number
+  timestamp: number
+  transactionId: number
+}
+
+export async function createTransactionPrices (params: CreateAllPricesFromTransactionInput): Promise<void> {
+  // Create USD price, if it does not already exist
+  const usdPrice = await prisma.price.upsert({
+    where: {
+      Price_timestamp_quoteId_networkId_unique_constraint: {
+        quoteId: USD_QUOTE_ID,
+        networkId: params.networkId,
+        timestamp: params.timestamp
+      }
+    },
+    create: {
+      value: params.values.usd,
+      timestamp: params.timestamp,
+      network: {
+        connect: {
+          id: params.networkId
+        }
+      },
+      quote: {
+        connect: {
+          id: USD_QUOTE_ID
+        }
+      }
+    },
+    update: {}
+  })
+  // Connect it with transaction, if not already connected
+  void await prisma.pricesOnTransactions.upsert({
+    where: {
+      priceId_transactionId: {
+        priceId: usdPrice.id,
+        transactionId: params.transactionId
+      }
+    },
+    create: {
+      transactionId: params.transactionId,
+      priceId: usdPrice.id
+    },
+    update: {}
+  })
+
+  // Create CAD price, if it does not already exist
+  const cadPrice = await prisma.price.upsert({
+    where: {
+      Price_timestamp_quoteId_networkId_unique_constraint: {
+        quoteId: CAD_QUOTE_ID,
+        networkId: params.networkId,
+        timestamp: params.timestamp
+      }
+    },
+    create: {
+      value: params.values.cad,
+      timestamp: params.timestamp,
+      network: {
+        connect: {
+          id: params.networkId
+        }
+      },
+      quote: {
+        connect: {
+          id: CAD_QUOTE_ID
+        }
+      }
+    },
+    update: {}
+  })
+  // Connect it with transaction, if not already connected
+  void await prisma.pricesOnTransactions.upsert({
+    where: {
+      priceId_transactionId: {
+        priceId: cadPrice.id,
+        transactionId: params.transactionId
+      }
+    },
+    create: {
+      transactionId: params.transactionId,
+      priceId: cadPrice.id
+    },
+    update: {}
+  })
+}
+
+export async function syncTransactionPrices (params: SyncTransactionPricesInput): Promise<void> {
+  if (appInfo.priceAPIURL === undefined) {
```

Comment:
I think environment variables are typically strings. This isn't triggering for me when I leave the PRICE_API_URL blank

Reply:
You probably left it as `PRICE_API_URL=`, which will set it to an empty string. If you don't mention the variable at all it will come as `undefined`.

Reply:
Ya that was it. All good then!